test/ng/controllerSpec.js

```diff
@@ -57,6 +57,19 @@ describe('$controller', function() {
       expect(ctrl instanceof BarCtrl).toBe(true);
     });
 
+    it('should return true when having an existing controller', function() {
+      var FooCtrl = function($scope) { $scope.foo = 'foo'; };
+
+      $controllerProvider.register({FooCtrl: FooCtrl});
+      expect($controllerProvider.has('FooCtrl')).toBe(true);
+    });
+
+    it('should return false when not having a non existing controller', function() {
+      var FooCtrl = function($scope) { $scope.foo = 'foo'; };
+
+      $controllerProvider.register({FooCtrl: FooCtrl});
+      expect($controllerProvider.has('BarCtrl')).toBe(false);
+    });
```

Comment:
Sorry, too much internals :grin:
`noop` is basically `angular.noop`, which is basically `function() {}` (a no-op function). You can use `noop` as a shorthand for an empty constructor function. For this functionality, you don't need to pass any arguments to the controller, nor does the controller need to have any logic.